c7n/resources/s3.py

```diff
@@ -375,6 +377,244 @@ def get_permissions(self):
         return self.permissions
 
 
+@filters.register('has-lifecycle')
+class HasLifecycle(Filter):
+    """Filter out S3 buckets that have lifecycle rules for
+    transitioning objects to different storage types."""
+
+    permissions = ('s3:GetLifecycleConfiguration', )
+
+    schema = type_schema('has-lifecycle',
+                         id={'type': 'string'},
```

Comment:
this should be marked as required, right?  Nothing will ever match without an `id` in the current code